### ISSUE TITLE
Latitude and Longitude - issue when it has to be entered in as a integer , can I multiple it by 10 (each lat and each long) to get decimal?

### ISSUE THREAD

**Issue by mventura00** [inria-UFF/VRPSolverEasy#26]
Hi developers,

I have a situation, where I am using the VRPsolvereasy, and entering in the 'latitude' and longitude for different counties in Pennsylvania that are closest to Pittsburg airport. I'm running into an issue, and wanted to give a proposed solution, and see what you thought of it? So , to give you an example , there are 2 counties (of the 26 counties designatied to Harrisburg airport), where they have this latitude and longitude:

'Indiana' county,       latitude = 40.68  , longitude = -79.1  , demand = 20
'Clearfield' county,   latitude = 40.95  , longitude = -78.56  , demand = 25

However, because VRPsolvereasy requires you to enter in the 'latitude' and 'longitude' as an integer, then they both have the same latitude and longitude (since you cant use decimals) so it looks like they're identical counties, which they're not, so I would have to enter it as (when I round it)

Indiana' county,       latitude = 41  , longitude = -79  , demand = 20
'Clearfield' county,   latitude = 41  , longitude = -79 ,  demand = 25

My question is: 
Can I multiple the latitude and longitude, each by 10 respectively (to capture the first digit after the decimal) and then later when I get the cost (lower bound), I will divide it by 10? 
Does that work? Does it make sense?
So I could potentially enter it into the input text file as:

Indiana' county,       latitude = 406  , longitude = -791  , demand = 20
'Clearfield' county,   latitude = 409  , longitude = -786 ,  demand = 25

In that way, I would be able to differentiate between the counties, and later,when I get the solution for cost, I'd divide by 10. 
What do you think?

**Comment by mventura00** [comment]
I just wanted to follow-up an update: so I did try to multiply the latitude and longitude each by 10, in order to get the 1 decimal point (and later I would divide the cost by 10), but for some reason the program didnt run well. Hmmm. I dont know, let me try again. The good news is that the depot (philly airport) only delivers to 10 cities, so it doesnt take that long to run fortunately. I will try again, and follow-up again!

**Comment by rrsadykov** [comment]
Dear @mventura00, 

VRPSolverEasy works directly with distances between points. It is up to the user to calculate these distances the way he likes. If VRPSolverEasy does not run well for some data, can you please send us the JSON file with input.

**Comment by mventura00** [comment]
> Dear @mventura00,
> 
> VRPSolverEasy works directly with distances between points. It is up to the user to calculate these distances the way he likes. If VRPSolverEasy does not run well for some data, can you please send us the JSON file with input.

Yes- I will send you the input file (in CSV file) of the 5 different input files (each for a different airport). Refer please to the columns under "Rounded" (for latitutde and longitutde), since it only takes integers .

Thanks for letting me know how VRPsolvereasy works directly between points. I will have to look into that too more... I saw taht there is a funcion called 'def comput_euclidean distance'... is this what you mean? its called a couple of times.

**Comment by mventura00** [comment]
For instance, this is what I mean - by how some counties end up having the same latitude and longitude (after you round to nearest integer). See tab called "State College airport"

[Final Assignment to 5 Airports.xlsx](https://github.com/user-attachments/files/17666822/Final.Assignment.to.5.Airports.xlsx)

**Comment by rrsadykov** [comment]
Transformation of input to the VRPSolverEasy model is the responsibility of the user, we cannot help you with that. 
When I was talking about JSON file, I was talking about the model export, see https://vrpsolvereasy.readthedocs.io/en/latest/Important%20notes/index.html#issues-and-debugging

**Comment by mventura00** [comment]
Thanks, I never heard of the JSON file before. Let me look into it.

**Here are some other ideas /issues I had regarding my problem:**
1. The input file are based in latitude and longitude degrees. This is an issue - because the fixed cost and variable cost for trucks that I put in , are based in $/ mile. I'm working to convert the degrees to miles, by doing : 
1 degree = 69.17 miles  , and then I can work with 'miles' instead, so then I am more consistent (with truck variable cost and fixed cost since it is also in miles. 
Thus,
Y miles = 69.17 * Latitude
X miles =COS(Latitude*0.0174533)*69.13*Longitude

![image](https://github.com/user-attachments/assets/edd54a49-2134-47d1-b716-49331907b66b)
in doing so, since everything is based off where the equator line is, then as a result, we are going to get thousands of miles (range of 2,000 - 4,000 miles+) in conversions. Instead, we can take the 'southern most' city of PA, which is Littlestown, PA, and then treating that as the x-axis = 0, and have all the other major city (in every county) based off of that , so then the range of miles is much more reasonable and on a lot smaller scale!
I did that and it looks like this (to give a sample/an idea): 
![image](https://github.com/user-attachments/assets/729a7bcc-535a-4b53-9294-240a1edc2175)

 Then, I'll need to  change the code (in python) in terms of how it is reading the 'input files ' (since we are using miles, instead of degrees) .  
I'll need to modify the function 'def compute_euclidean_distance'
So that I give it the 'i' and the 'j' , as it is checking the distance between 'i' and 'j' (for python file: HFVRP python file)

def compute_euclidean_distance(x_i, y_i, x_j, y_j, number_digit=3):
    """Compute the euclidean distance between 2 points from graph"""
    return round(math.sqrt((x_i - x_j)**2 + (y_i - y_j)**2), number_digit)

 **Thus, I need to make 2 changes:**

1. Read the data file and save it in a different format. A new list (a pair of cities) and a distance between them, for every city (which I do have already, but need to organize it better in this format)
2. Modify the section of the code  for 'def compute_euclidean_distance' so it reads the data the way that I saved it!

**Comment by mventura00** [comment]
My Google API distance matrix  were in  that I based my 'assignments' from before are attached: these are all the distance for all 5 airports, from all 67 counties of PA.
[All 5 airport- calc the closest airport.xlsx](https://github.com/user-attachments/files/17708561/All.5.airport-.calc.the.closest.airport.xlsx)
[Final Assignment to 5 Airports.xlsx](https://github.com/user-attachments/files/17708566/Final.Assignment.to.5.Airports.xlsx)

**Comment by rrsadykov** [comment]
I close the issue as there are no further messages from the issue opener for several months.
